docs/06-contributors/999-rfcs/000-stories/story-12b.md

```diff
@@ -0,0 +1,24 @@
+# User Story 12b - bring jsii in preflight
+
+> **Status**: Expected released on 2023/02/02
+
+We want to be able to bring a jsii module into wing, this story focuses on the smallest step,
+bringing the code so it can be used in preflight.
+
+Here is the expected test that should work in the end of this sprint:
+
+```ts (wing)
+// Prerequisite
+// npm install --save cdk-constants
+
+bring cloud;
+bring "cdk-constants" as cdk_constants;
+
+// capture jsii data in preflight
+let text_from_jsii = cdk_constants.ManagedPolicies.AWS_LAMBDA_BASIC_EXECUTION_ROLE;
+
+new cloud.Function(inflight (s: str): str => {
+  assert("service-role/AWSLambdaBasicExecutionRole" == text_from_jsii);
+}) as "test:able to capture jsii in preflight";
```

Comment:
This looks correct to me

Reply:
In the future, anyways. While working on jsii imports I remembered that we don't support static class members, and it turns out this is what the constants from `cdk-constants` are (`ManagedPolicies` is actually a class, not an enum, go figure...) -- so that particular example might not work. But I'll find another example that will work - stay tuned.

Reply:
Thanks @Chriscbr , I am tuned

Reply:
@Chriscbr, any findings?

Reply:
So, it's a pitifully sad example, but in my current "bring jsii" PR I've got this example working:

```js
bring cloud;
bring "jsii-code-samples" as stuff;

let hello = new stuff.HelloWorld();
let greeting = hello.say_hello("wingnuts");

new cloud.Function(inflight (m: str): str => {
  assert(greeting == "Hello, wingnuts");
}) as "test:say_hello";
```

([Here](https://github.com/aws-samples/jsii-code-samples/blob/main/lib/index.ts) is the entire source code of the jsii-code-samples library).

I found a lot of issues with trying to import CDKTF resources, from missing variadics to missing interfaces, though it's possible we could decide to just patch over them by skipping importing the types or importing them as "any" for now (without exposing them to users maybe). When importing cdk8s (a pretty small and self-contained CDK library without many dependencies) I ran into one issue with importing `constructs.Node`, though I didn't have time to investigate it yet. I will continue investigating - let me know if this example is an ok for now. 🙂

Reply:
This is awesome @Chriscbr!

Regarding gaps in our type system, can you create an issue with a list of missing features, so we can prioritize for beta? I really think we need to be able to import cdktf libraries for beta...

Reply:
@staycoolcall911 FYI 


I think we need to decide whether we are prioritizing `bring jsii` (into inflight) or maybe going with `bring untyped` (as an alternative) vs `bring CDKTF` resources for the short term. 

@Chriscbr, would love to hear your opinion here

Reply:
Scheduled a meeting

Reply:
@ekeren, @eladb, @Chriscbr 
I think we have the following issues at this point:
1. [Jsii variadic parameter support](https://github.com/winglang/wing/issues/397), which depends on  [Function Declaration: Variadic Arguments](https://github.com/winglang/wing/issues/125).
1. [jsii importer fails when trying to import an interface](https://github.com/winglang/wing/issues/1024), which depends on [Interface Declaration](https://github.com/winglang/wing/issues/123).